### PR TITLE
syncthingtray: install app bundle in `/Applications`

### PR DESCRIPTION
**Pull request by davidkna**
## Description of changes

The app bundle was being installed to `$out/bin`. By moving it into `$out/Applications` the app bundle becomes more easily accessible via `/Applications/Nix Apps/` etc.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I built the package, verified that it now gets added to the applications folder, and launched the app bundle from there.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by doronbehar** [comment]
I pushed another small attribute rename that requested in a different PR.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -35239,18 +35239,18 @@ with pkgs;
...
     jsSupport = false;
     kioPluginSupport = false;
     plasmoidSupport = false;
     systemdSupport = true;
   };
-  syncthingtray-qt6 = kdePackages.callPackage ../applications/misc/syncthingtray {
-    # renamed in KF5 -> KF6
-    plasma-framework = kdePackages.libplasma;
-  };
+  syncthingtray-qt5 = libsForQt5.callPackage ../applications/misc/syncthingtray { };
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:35253]: Is this really needed?

Comment (doronbehar) [reply pkgs/top-level/all-packages.nix:35253]: TBH I haven't thought about a lot, but I think there are many people out there who don't use KDE and don't have a lot of QT packages, and the packages they use don't support qt6 yet, and they'd wish to not depend on qt6 if they don't have to. OTH, adding a new qt5 attribute could be going against progress, so I won't mind removing it.

Comment (dotlambda) [reply pkgs/top-level/all-packages.nix:35253]: Let's remove it. There are many other packages that only use Qt 5 despite supporting both.

**Comment by davidkna** [comment]
I updated my original commit to also add the binary in the app bundle to `$out/bin` to make it available on `$PATH` like on other platforms.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/top-level/all-packages.nix:35253]: > Let's remove it. There are many other packages that only use Qt 5 despite supporting both.

:+1:.

**Comment by doronbehar** [comment]
Will merge when CI is green.

**Comment by doronbehar** [comment]
CI was fully green before, and there was no change other then the `syncthingtray-qt5` removal. So I'm merging now although darwin CI is still yellow.

**Comment by tschai-yim** [comment]
Setting `syncthingtray-qt6` as default broke Syncthing tray for my GNOME system. Seems some library is missing:
```
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
QApplication: invalid style override 'adwaita-dark' passed, ignoring it.
        Available styles: Windows, Fusion
```
I'll see if I can make a patch

**Comment by doronbehar** [comment]
Thanks for the info @tschai-yim . I haven't updated my system yet with this patch. Should be fixed by: https://github.com/NixOS/nixpkgs/pull/294935

**Pull request merged**